backend/uclapi/uclapi/ourcelery.py

```diff
@@ -17,9 +14,15 @@
 class Celery(celery.Celery):
     def on_configure(self):
         if os.environ.get("SENTRY_DSN") is not None:
-            client = raven.Client(os.environ.get("SENTRY_DSN"))
-            raven_register_logger_signal(client)
-            raven_register_signal(client)
+            import sentry_sdk
+            from sentry_sdk.integrations.django import DjangoIntegration
+
+            sentry_sdk.init(
+                dsn=os.environ.get("SENTRY_DSN"),
+                integrations=[DjangoIntegration()],
+                traces_sample_rate=0.01,
+                send_default_pii=True
```

Comment:
We need the `before_send` keyword argument to be filled as well: https://docs.sentry.io/platforms/python/configuration/filtering/#using-beforesend

This is particularly important for us as we have the legacy of putting confidential information as a query parameter, which we should prevent from being stored on other servers. So make sure we strip tokens, client_id/state etc.

Also strip anything else that we don't need for debugging but is sensitive.

Reply:
It turns out that sentry sends quite a bit more information than I was expecting, some in several different forms (I counted the URL with get parameters provided in at least 5 different nodes of the tree). Sentry has some server side scrubbing but ideally we would not send anything sensitive.

This is made harder by the event tree being very variable and has resulted in this recursive explorer checking for dict/list entries. We then check against a list of known keys which contain sensitive information and remove those entries. For the URL entries and some trace backs (the get parameters are passed in as arguments so sometimes shown in the stack trace), this approach uses regex's to match the possibly generated tokens, client ids, and client secrets. I am not sure of a better solution (without regex) which won't redact too much debugging information.